src/mon/Elector.cc

```diff
@@ -762,7 +762,7 @@ void Elector::notify_rank_removed(unsigned rank_removed, unsigned new_rank)
      In the case where we are removing the highest rank,
      we erase the removed rank from all sets.
    */
-  if (rank_removed < paxos_size()) {
+  if (static_cast<unsigned int>(rank_removed) < paxos_size()) {
```

Comment:
consider using the c++20 function [cmp_less()](https://en.cppreference.com/w/cpp/utility/intcmp) which handles signed/unsigned comparisons transparently

Reply:
Try using this approach to submit the new changes.